### DISCUSSION TITLE
FastAPI doesn't format datetime.datetime output to ISO 8601 as documented

### DISCUSSION THREAD

**Discussion by ferulisses** [fastapi/fastapi#6228] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pydantic import BaseModel
from fastapi import FastAPI
import datetime


class TestOutput(BaseModel):
    date: datetime.datetime

def app():
    fast_api_app = FastAPI()

    @fast_api_app.get("/", response_model=TestOutput)
    def test():
        return {'date': datetime.datetime.utcnow()}

    return fast_api_app
```

### Description

Opening the web browser, I got the response:
```
{
  "date": "2022-06-14T19:25:55.821177"
}
```
The timezone information is missing, I expected "+00:00" in the end 

The docs indicates that I should receive an ISO 8601 formatted date, that includes timezone
https://fastapi.tiangolo.com/tutorial/extra-data-types/

Even the example provided in OpenAPI (/docs) have the timezone information as the follow:
```
{
  "date": "2022-06-14T19:25:55.846Z"
}
```

### Operating System

Linux

### FastAPI Version

0.75.0

### Python Version

3.8.10

**Comment by zoliknemet** [comment, marked as answer]
Hi, the problem is in this datetime you are returning as response
`datetime.utcnow()` is NAIVE timestamp - it does not know what timezone it is in

better way is to do it like this
```python
import pytz # this extra import
@fast_api_app.get("/", response_model=TestOutput)
def test():
    return {"date": datetime.datetime.now(tz=pytz.utc)} # datetime that is timezone aware - its timezone is UTC
```

datetime.utcnow() could be a little bit missleading...

EDIT: play a little bit with datetime in python
```python
from datetime import datetime
import pytz

now = datetime.now()
print("My local time but timezone unaware - naive time")
print(now)
print(now.isoformat())
print(now.tzinfo)

now_tz_london = datetime.now(pytz.timezone("Europe/London"))
print("London time with timezone - aware time")
print(now_tz_london)
print(now_tz_london.isoformat())
print(now_tz_london.tzinfo)

utc_now = datetime.utcnow()
print("Utc but timezone unaware - naive time")
print(utc_now)
print(utc_now.isoformat())
print(utc_now.tzinfo)
```

**Comment by ferulisses** [comment]
Thanks for your response, I understood it.

There is any way to force the timezone on output, like Redoc and SwaggerUI do? Do you think that it's a good idea?

**Comment by slackner** [comment]
You can add a custom pydantic type to achieve that. See the following code snippet as an example:

```python3
from pydantic import BaseModel
from pydantic.datetime_parse import parse_datetime
from fastapi import FastAPI
import datetime

class UTCDatetime(datetime.datetime):
    @classmethod
    def __get_validators__(cls):
        yield parse_datetime  # default Pydantic behavior
        yield cls.validate

    @classmethod
    def validate(cls, value) -> str:
        if value.tzinfo is None:
            return value.replace(tzinfo=datetime.timezone.utc)

        # Convert to UTC
        return value.astimezone(datetime.timezone.utc)
        # or alternatively, return the original value:
        # return value

class TestOutput(BaseModel):
    date: UTCDatetime

def app():
    fast_api_app = FastAPI()

    @fast_api_app.get("/", response_model=TestOutput)
    def test():
        return {'date': datetime.datetime.utcnow()}

    return fast_api_app
```

As long as you are careful where you are using timezone-aware and non-timezone-aware timestamps, using such an approach is fine.

(Note: I'm using this myself, but for input parameters instead of output parameters. The type hint `datetime.datetime` accepts both timezone-aware and non-timezone-aware timestamps, and comparing both in the API endpoint can lead to an exception otherwise.)

**Comment by slackner** [comment]
A small side note, there is no need to use `pytz` to generate timezone aware timestamps. You can also use:

```python3
datetime.datetime.now(datetime.timezone.utc)
```

**Reply by edward-b-1** [reply]
Just for the benefit of future readers.

If you import

```
from datetime import datetime
from datetime import timezone
```

then you can do

```
datetime.now(timezone.utc)
```

Also just to note, if you need something else form the datetime package, you will have to explicitly import it, since the name `datetime` now means the `datetime` type from the `datetime` package. Look up how the import system and modules/packages work if you're not familiar with this.

**Comment by ferulisses** [comment]
Thanks, great solution.
